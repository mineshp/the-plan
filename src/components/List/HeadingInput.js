import React from 'react';
import PropTypes from 'prop-types';
import { Button, Divider, Form, Grid, Input } from 'semantic-ui-react';

const HeadingInput = ({ headings, handleAddColumn, handleRemoveColumn, handleHeaderInputChange }) => {
    const headerRows = [];
    headings.map((heading) => {
        const headingID = heading.id; // eslint-disable-line no-underscore-dangle
        return (
            headerRows.push(
                <Grid.Row key={headingID}>
                    <Grid.Column>
                        <Form.Field>
                            <Input
                                placeholder="Header Name..."
                                name={heading.name}
                                id={headingID}
                                key={headingID}
                                defaultValue={heading.name}
                                onChange={handleHeaderInputChange}
                            />
                        </Form.Field>
                    </Grid.Column>
                    <Grid.Column>
                        <Button
                            icon="trash"
                            content="Remove"
                            labelPosition="left"
                            id={headingID}
                            color="pink"
                            type="button"
                            onClick={handleRemoveColumn}
                            floated="left"
                        />
                    </Grid.Column>
                </Grid.Row>
            )
        );
    });

    return (
        <div>
            <Grid columns={2} divided>
                <Divider horizontal>ASSIGN LIST HEADINGS</Divider>
                {headerRows}
                <Divider />
                <Button fluid type="button" color="blue" onClick={handleAddColumn}>Add Column</Button>
            </Grid>
        </div>
    );
};

HeadingInput.propTypes = {
    headings: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string,
            name: PropTypes.string
        }).isRequired
    ).isRequired,
    handleAddColumn: PropTypes.func.isRequired,
    handleRemoveColumn: PropTypes.func.isRequired,
    handleHeaderInputChange: PropTypes.func.isRequired
};

export default HeadingInput;
