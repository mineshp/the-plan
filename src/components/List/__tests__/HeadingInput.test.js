import React from 'react';
import renderer from 'react-test-renderer';
import HeadingInput from '../HeadingInput';
import { mockHeadingsData } from '../../../helpers/test/testData/listData';

const mockHeadings = mockHeadingsData();

const mockHandleAddColumn = jest.fn();
const mockHandleRemoveColumn = jest.fn();
const mockHandleHeaderInputChange = jest.fn();

describe('HeadingInput', () => {
    it('renders column headings in a grid correctly', () => {
        const tree = renderer.create(
            <HeadingInput
                headings={mockHeadings}
                handleAddColumn={mockHandleAddColumn}
                handleRemoveColumn={mockHandleRemoveColumn}
                handleHeaderInputChange={mockHandleHeaderInputChange}
            />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
