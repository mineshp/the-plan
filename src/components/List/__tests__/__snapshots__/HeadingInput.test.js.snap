// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingInput renders column headings in a grid correctly 1`] = `
<div>
  <div
    className="ui divided three column grid"
  >
    <div
      className="ui horizontal divider"
    >
      COLUMNS
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              disabled={undefined}
              id="000001"
              name="A"
              onChange={[Function]}
              placeholder="Header Name..."
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              disabled={undefined}
              id="000001"
              name={undefined}
              onChange={[Function]}
              placeholder="Position..."
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <button
          className="ui pink icon left labeled button"
          disabled={undefined}
          id="000001"
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="button"
        >
          <i
            aria-hidden="true"
            className="trash icon"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="column"
      >
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              disabled={undefined}
              id="000002"
              name="B"
              onChange={[Function]}
              placeholder="Header Name..."
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <div
          className="field"
        >
          <div
            className="ui input"
          >
            <input
              disabled={undefined}
              id="000002"
              name={undefined}
              onChange={[Function]}
              placeholder="Position..."
              tabIndex={undefined}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="column"
      >
        <button
          className="ui pink icon left labeled button"
          disabled={undefined}
          id="000002"
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
          type="button"
        >
          <i
            aria-hidden="true"
            className="trash icon"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      className="ui divider"
    />
    <button
      className="ui fluid button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="button"
    >
      Add Column
    </button>
  </div>
</div>
`;
